File: src/2859.sum-of-values-at-indices-with-k-set-bits.3093/solution.ts
function count1 (n:number) {
    let r = 0;
    while (n) {
        n &= (n - 1);
        r++;
    }
    return r;
}

function sumIndicesWithKSetBits (nums: number[], k: number): number {
    let res = 0;
    for (let i = 0; i < nums.length; i++) {
        if (count1(i) === k) {
            res += nums[i];
        }
    }

    return res;
}

File: src/2859.sum-of-values-at-indices-with-k-set-bits.3093/solution.java
class Solution {
    public int sumIndicesWithKSetBits(List<Integer> nums, int k) {
        int result = 0;
        for (int i = 0; i < nums.size(); i++) {
            int n = i;
            int cnt = 0;
            while (n != 0) {
                cnt++;
                n &= (n - 1);
            }

            if (cnt == k) {
                result += nums.get(i);
            }
        }
        return result;
    }
}